docgen/docgen.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"path/filepath"
+	"strings"
+
+	"github.com/golang/protobuf/proto"
+	google_protobuf "github.com/golang/protobuf/protoc-gen-go/descriptor"
+	plugin_go "github.com/golang/protobuf/protoc-gen-go/plugin"
+
+	"github.com/gunk/gunk/docgen/extract"
+	pg "github.com/gunk/gunk/plugin"
+)
+
+func main() {
+	if err := pg.RunPlugin(new(plugin)); err != nil {
+		log.Fatal(err)
+	}
+}
+
+type plugin struct{}
+
+func (p *plugin) Generate(req *plugin_go.CodeGeneratorRequest) (*plugin_go.CodeGeneratorResponse, error) {
+	// For now, we only care about all.proto
+	var source *google_protobuf.FileDescriptorProto
+	for _, f := range req.GetProtoFile() {
+		if strings.Contains(f.GetName(), "all.proto") {
+			source = f
+			break
+		}
+	}
+
+	if source == nil {
+		return nil, fmt.Errorf("no file to generate")
+	}
+
+	name := proto.String(filepath.Join(filepath.Dir(source.GetName()), "messages.pot"))
```

Comment:
nit: this would be a bit clearer like:

```
dest := filepath.Join(filepath.Dir(source.GetName()), "messages.pot")
...
Name: proto.string(dest)
```

Reply:
(in particular, "name" doesn't really say much, and I had to read the rest of the code to realise that it was the *desination* path name)